tpm/tpm.go

```diff
@@ -35,26 +37,60 @@ type Closer interface {
 // when using the kernel resource manager.
 func Open(path string) (Closer, error) {
 	switch {
-	case isDevNode(path, "tpmrm"):
+	case IsDevNode(path, DevNodeManaged):
 		return linuxtpm.Open(path)
-	case isDevNode(path, "tpm"):
+	case IsDevNode(path, DevNodeUnmanaged):
 		slog.Warn("direct use of the TPM can lead to resource exhaustion, use a TPM resource manager instead")
 		return linuxtpm.Open(path)
 	default:
 		return nil, fmt.Errorf("unsupported TPM device path: %s", path)
 	}
 }
 
-func isDevNode(path, kind string) bool {
-	if strings.Contains(kind, "/") {
-		return false
+// DevNodeKind enumerates managed and unmanaged TPM dev node kinds.
+type DevNodeKind uint8
+
+// PathPrefix returns the standard unix /dev node path without the numerical
+// index.
+func (n DevNodeKind) PathPrefix() string {
+	switch n {
+	case DevNodeUnmanaged:
+		return "/dev/tpm"
+	case DevNodeManaged:
+		return "/dev/tpmrm"
 	}
-	prefix := "/dev/" + kind
+	panic("invalid TPM dev node kind")
```

Comment:
I wouldn't panic in a library

Reply:
My rule of thumb is "panic when the code should be unreachable" so that future changes to the code that break an invariant are caught faster. This panic is basically just there to catch a non-exhaustive switch statement on an `enum`. If the Go compiler had support for this invariant, the panic would not be necessary. As is, the panic is safe.